indico/modules/events/persons/controllers.py

```diff
@@ -347,3 +358,51 @@ def _process(self):
             tpl = get_template_module('events/persons/management/_person_list_row.html')
             return jsonify_data(html=tpl.render_person_row(person_data, bool(self.event.registration_forms)))
         return jsonify_form(form)
+
+
+class RHEventPersonSearch(RHPersonsBase):
+    def _search_event_persons(self, exact=False, **criteria):
+        criteria = {key: v for key, value in criteria.items() if (v := value.strip())}
+
+        if not criteria:
+            return []
+
+        query = EventPerson.query.distinct(EventPerson.id).filter(EventPerson.event_id == self.event.id)
+
+        query = query.filter(
+            or_(
+                EventPerson.abstract_links.any(AbstractPersonLink.abstract.has(~Abstract.is_deleted)),
+                EventPerson.contribution_links.any(ContributionPersonLink.contribution.has(~Contribution.is_deleted)),
+                EventPerson.event_links.any(),
+                EventPerson.subcontribution_links.any(
+                    SubContributionPersonLink.subcontribution.has(~SubContribution.is_deleted)),
+                EventPerson.subcontribution_links.any(~SubContribution.contribution.has(Contribution.is_deleted)),
```

Comment:
This needs to be one condition - otherwise having a link to a deleted subcontribution of a non-deleted contribution would work.

I think this works (but untested):

```suggestion
                EventPerson.subcontribution_links.any(
                    SubContributionPersonLink.subcontribution.has(
                        ~SubContribution.is_deleted,
                        SubContribution.contribution.has(~Contribution.is_deleted)
                    )
                ),
```